src/function/vector_list_operation.cpp

```diff
@@ -140,31 +147,32 @@ std::vector<std::unique_ptr<VectorOperationDefinition>>
 ListConcatVectorOperation::getDefinitions() {
     std::vector<std::unique_ptr<VectorOperationDefinition>> result;
     auto execFunc = BinaryListExecFunction<ku_list_t, ku_list_t, ku_list_t, operation::ListConcat>;
-    auto bindFunc = [](const std::vector<DataType>& argumentTypes, FunctionDefinition* definition,
-                        DataType& actualReturnType) {
-        if (argumentTypes[0] != argumentTypes[1]) {
-            throw BinderException(getListFunctionIncompatibleChildrenTypeErrorMsg(
-                LIST_CONCAT_FUNC_NAME, argumentTypes[0], argumentTypes[1]));
+    auto bindFunc = [](const binder::expression_vector& argumentTypes,
+                        FunctionDefinition* definition, DataType& actualReturnType) {
+        if (argumentTypes[0]->getDataType() != argumentTypes[1]->getDataType()) {
+            throw BinderException(
+                getListFunctionIncompatibleChildrenTypeErrorMsg(LIST_CONCAT_FUNC_NAME,
```

Comment:
should we abstract not only error msg but also the if statement. It seems 152 - 155 is being reused for multiple list bind functions.

Reply:
I think the if/else statement is different in each case:
E.g. contains: needs the child type of 1st parameter equals to the second parameter type.
Whereas prepends: needs the type of the 1st parameter equals to the child type of the second parameter.